src/Generator/EnumGenerator/Cases/PureCases.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Laminas\Code\Generator\EnumGenerator\Cases;
+
+use Laminas\Code\Generator\EnumGenerator\CasesInterface;
+
+use function array_values;
+
+final class PureCases implements CasesInterface
+{
+    /** @var list<string> */
+    private array $cases;
+
+    /**
+     * @param list<string> $cases
+     */
+    private function __construct(array $cases)
+    {
+        $this->cases = $cases;
+    }
+
+    public function getBackedType(): ?string
+    {
+        return null;
```

Comment:
If we remove the interface (as suggested below), and only use `PureCases|BackedCases` in our system, this nullability goes away